### PR TITLE
Account for Across minDeposit when bridging funds

### PR DESCRIPTION
**Pull request by kevincheng96**
This PR adds logic to account for the `minDeposit` (e.g. min bridge amount) when using the Across bridge. If there are not enough funds on a chain to meet the `minDeposit` amount, the QuarkBuilder will not attempt to bridge from that chain.

We also update the logic for `getTotalAvailableBalance` to calculate the total available balance only on the destination chain. This primarily affects the amount that is used for max actions (e.g. transfer max).

### CODE REVIEW

File: src/builder/Accounts.sol [hunk 1]
```diff
@@ -355,12 +356,22 @@ library Accounts {
...
+        // If payment fees are higher than the unbridgeable balance, it means that we cannot use the unbridgeable
+        // funds to pay for the fees, so fees need to be paid on the destination chain
+        if (paymentFees > unbridgeableBalance) {
+            if (balance - totalBridgeFees - unbridgeableBalance < paymentFees) {
+                return 0;
+            }
+            return balance - totalBridgeFees - paymentFees - unbridgeableBalance;
+        } else {
+            return balance - totalBridgeFees - unbridgeableBalance;
+        }
```
Comment (kevincheng96) [inline src/builder/Accounts.sol:374]: @mykelp I changed this logic slightly from what we originally discussed (`balance - totalBridgeFees - Math.max(paymentFees, unbridgeableBalance)`) to properly handle the edge case where `paymentFees > unbridgeableBalance`, meaning fees cannot be paid from the unbridgeable chains.

File: src/builder/actions/Actions.sol [hunk 2]
```diff
@@ -576,11 +576,18 @@ library Actions {
...
                     );
 
-                    amountLeftToBridge -= outputAmount;
-                    totalBridgeFees += (inputAmount - outputAmount);
-
-                    List.addAction(actions, action);
-                    List.addQuarkOperation(quarkOperations, operation);
+                    // If outputAmount is 0, it means we are not able to bridge the funds
+                    if (outputAmount == 0) {
+                        unbridgeableBalance += Accounts.getBalanceOnChain(
```
[the remark this answers is not in the document]
Comment (mykelp) [reply src/builder/actions/Actions.sol:581]: Right now I guess that would be accounted as fees. We'd have to explicitly return the fees from `bridgeAsset` if we wanted to clarify

[the remark this answers is not in the document]
Comment (kevincheng96) [reply src/builder/actions/Actions.sol:581]: There is some [logic](https://github.com/legend-hq/legend-scripts/blob/ee855763d6771753e90d20502b3ca25f5eae6db8/src/builder/QuarkBuilderBase.sol#L337-L340) in `constructOperationsAndActions` that returns an error if we aren't able to bridge the total requested amount.

File: test/Acceptance/Tests/AcceptanceTests/TransferTests.swift [hunk 3]
```diff
@@ -238,4 +238,107 @@ let transferTests: [AcceptanceTest] = [
...
+            .single(
+                    .multicall([
+                        // Only 10 USDC is available to transfer since payment has to be made on Arbitrum
+                        // due to unbridgeable funds on Base
+                        .transferErc20(tokenAmount: .amt(10, .usdc), recipient: .bob, network: .arbitrum),
+                        .quotePay(payment: .amt(40, .usdc), payee: .stax, quote: .basic)
+                ])
+            )
+        )
+    )
```
Comment (kevincheng96) [inline test/Acceptance/Tests/AcceptanceTests/TransferTests.swift:343]: These are the main edge cases

**Comment by kevincheng96** [comment]
Ok this should be ready for a re-review. Commit 152a544c8aac1a65199d1ba26eafafa1ff114ddf should make the logic for calculating max intent amounts more logical.

**Review by mykelp — APPROVED** [review]

File: src/builder/QuarkBuilderBase.sol [hunk 4]
```diff
@@ -541,15 +539,16 @@ contract QuarkBuilderBase {
...
                     chainAccountsList,
                     payment,
+                    amountsOnDst,
                     bridgeFees,
                     List.addUniqueUint256(chainIdsInvolved, intent.chainId),
                     intent.assetSymbol
                 );
 
-                intent.amount = repayAmount < totalAvailable ? type(uint256).max : totalAvailable;
+                intent.amount = repayAmount <= totalAvailableOnDst ? type(uint256).max : totalAvailableOnDst;
```
[the remark this answers is not in the document]
Comment (kevincheng96) [reply src/builder/QuarkBuilderBase.sol:551]: Discussed offline. Now that we calculate `totalAvailableOnDst ` instead of the `totalAvailable` on all chains, the `repayAmount` and `totalAvailableOnDst` will always be equals in the case where the user has enough liquidity. In that happy path, we want the intent amount to be `uint256.max` instead of an exact amount.

**Comment by kevincheng96** [comment]
Looks like the one failing Foundry test is unrelated and likely related to the latest nightly (unstable) release of Foundry, so I will ignore for now.

```
[FAIL: call didn't revert at a lower depth than cheatcode call depth] testFindPriceFeedPathNoMatch() (gas: 1523223)
```
